test/tbb/test_overwrite_node.cpp

```diff
@@ -183,6 +184,163 @@ void test_deduction_guides() {
+                    return 0;
+                });
+
+            tbb::flow::make_edge(buffer, function);
+
+            buffer.try_put(start_message); // Occupies concurrency of function
+
+            buffer.try_put_and_wait(wait_message);
+
+            CHECK_MESSAGE(processed_items.size() == 2, "only the start_message and wait_message should be processed");
+            std::size_t check_index = 0;
+            CHECK_MESSAGE(processed_items[check_index++] == start_message, "unexpected start_message processing");
+            CHECK_MESSAGE(processed_items[check_index++] == wait_message, "unexpected wait_message processing");
+
+            g.wait_for_all();
+
+            CHECK_MESSAGE(processed_items[check_index++] == new_message, "unexpected new_message processing");
+            CHECK(check_index == processed_items.size());
+        });
+    }
+    // Test reserve
+    {
+        tbb::task_arena arena(1);
+
+        arena.execute([&] {
+            std::vector<int> processed_items;
+
+            tbb::flow::graph g;
+            tbb::flow::overwrite_node<int> buffer(g);
+            tbb::flow::limiter_node<int, int> limiter(g, 1);
+            int start_message = 0;
+            int new_message = 1;
+
+            using function_node_type = tbb::flow::function_node<int, int, tbb::flow::rejecting>;
+
+            function_node_type function(g, tbb::flow::serial,
+                [&](int input) {
+                    if (input == wait_message) {
+                        buffer.try_put(new_message);
+                    }
+
+                    // Explicitly clean the buffer to prevent infinite try_get by the function_node
+                    if (input == new_message) {
+                        buffer.clear();
+                    }
+
+                    processed_items.emplace_back(input);
+                    limiter.decrementer().try_put(1);
+                    return 0;
+                });
+
+            tbb::flow::make_edge(buffer, limiter);
+            tbb::flow::make_edge(limiter, function);
+
+            buffer.try_put(start_message); // Occupies concurrency of function
+
+            buffer.try_put_and_wait(wait_message);
+
+            CHECK_MESSAGE(processed_items.size() == 2, "only the start_message and wait_message should be processed");
+            std::size_t check_index = 0;
+            CHECK_MESSAGE(processed_items[check_index++] == start_message, "unexpected start_message processing");
+            CHECK_MESSAGE(processed_items[check_index++] == wait_message, "unexpected wait_message processing");
+
+            g.wait_for_all();
+
+            CHECK_MESSAGE(processed_items[check_index++] == new_message, "unexpected new_message processing");
+            CHECK(check_index == processed_items.size());
+        });
+    }
+    // Test explicit clear
+    {
+        tbb::flow::graph g;
+        tbb::flow::overwrite_node<int> buffer(g);
+
+        int processed_item = -42;
```

Comment:
I'd suggest to still use a vector for processed_items. It is possible here that you could receive more than 1 copy of wait_message in the function_node and the test would still pass. Instead if you keep it as a vector and check the size, you are sure that you only received the value once.

Reply:
Done